File: 배열_2차원/프로그래머스Lv2-n^2 배열 자르기.js
const judgeValue = (index, n) => {
  const [a, b] = [Math.floor(index / n), index % n];
  return a < b ? b + 1 : a + 1;
};

function solution(n, left, right) {
  const length = right - left + 1;
  const slicedIndexArr = Array.from({ length }, (v, i) => i + left);
  return slicedIndexArr.map((index) => judgeValue(index, n));
}
// 정확성  테스트
// 테스트 1 〉	통과 (17.14ms, 44.3MB)
// 테스트 2 〉	통과 (12.32ms, 48.9MB)
// 테스트 3 〉	통과 (12.81ms, 48.8MB)
// 테스트 4 〉	통과 (0.32ms, 33.6MB)
// 테스트 5 〉	통과 (0.23ms, 33.4MB)
// 테스트 6 〉	통과 (15.39ms, 43MB)
// 테스트 7 〉	통과 (16.22ms, 42.9MB)
// 테스트 8 〉	통과 (12.22ms, 44.1MB)
// 테스트 9 〉	통과 (14.15ms, 45.2MB)
// 테스트 10 〉	통과 (21.22ms, 44.1MB)
// 테스트 11 〉	통과 (13.67ms, 45.1MB)
// 테스트 12 〉	통과 (15.73ms, 48.8MB)
// 테스트 13 〉	통과 (16.52ms, 47.9MB)
// 테스트 14 〉	통과 (21.57ms, 48.3MB)
// 테스트 15 〉	통과 (18.36ms, 49.9MB)
// 테스트 16 〉	통과 (18.83ms, 50.9MB)
// 테스트 17 〉	통과 (14.22ms, 47.5MB)
// 테스트 18 〉	통과 (15.52ms, 52.1MB)
// 테스트 19 〉	통과 (25.83ms, 49.7MB)
// 테스트 20 〉	통과 (19.71ms, 44.8MB)

File: 배열_2차원/프로그래머스Lv2-n^2 배열 자르기.py
def judge_value(index, n):
    a, b = divmod(index, n)
    if a < b:
        return b + 1
    return a + 1

def solution(n, left, right):
    sliced_index_arr = [x for x in range(left, right + 1)]
    return [judge_value(x, n) for x in sliced_index_arr]

"""
정확성  테스트
테스트 1 〉	통과 (19.70ms, 16.2MB)
테스트 2 〉	통과 (24.15ms, 19.2MB)
테스트 3 〉	통과 (23.20ms, 19.2MB)
테스트 4 〉	통과 (0.02ms, 10.2MB)
테스트 5 〉	통과 (0.02ms, 10.2MB)
테스트 6 〉	통과 (22.35ms, 18.4MB)
테스트 7 〉	통과 (24.09ms, 19MB)
테스트 8 〉	통과 (20.64ms, 17.5MB)
테스트 9 〉	통과 (23.50ms, 18.9MB)
테스트 10 〉	통과 (24.47ms, 18.4MB)
테스트 11 〉	통과 (31.21ms, 18.5MB)
테스트 12 〉	통과 (21.00ms, 17.9MB)
테스트 13 〉	통과 (23.40ms, 18.2MB)
테스트 14 〉	통과 (23.29ms, 18.1MB)
테스트 15 〉	통과 (21.19ms, 17.9MB)
테스트 16 〉	통과 (22.75ms, 18.5MB)
테스트 17 〉	통과 (25.90ms, 18.5MB)
테스트 18 〉	통과 (25.37ms, 19.2MB)
테스트 19 〉	통과 (24.90ms, 18.6MB)
테스트 20 〉	통과 (22.12ms, 18MB)
"""